scripts/optimize.py

```diff
@@ -28,29 +35,124 @@
+    emission_df_raw = scalars.loc[scalars["carrier"] == "emission"].set_index(
+        "var_name"
+    )
+    emission_df = drop_values_by_keyword(emission_df_raw)
 
     # return None if no emission limit is given ('None' or entry missing)
     if emission_df.empty:
-        print("No emission limit set.")
+        print("No emission limit will be set.")
         return None
     else:
-        limit = emission_df.at["emission_limit", "var_value"]
-        print(f"Emission limit set to {limit}.")
+        limit = emission_df.at[EMISSION_LIMIT, "var_value"]
+        print(f"Emission limit will be set to {limit}.")
         return limit
 
 
+def get_electricity_gas_relations(scalars):
+    r"""
+    Gets electricity/gas relations from scalars. Returns None if no relations are given.
+
+    Returns
+    -------
+    pd.DataFrame
+        Contains rows of scalars with 'var_name' `EL_GAS_RELATION`
+    If no factor is given returns None.
+    """
+    relations_raw = scalars.loc[scalars.var_name == EL_GAS_RELATION]
+    # drop relations that are None
+    relations = drop_values_by_keyword(relations_raw)
+    if relations.empty:
+        print("No gas electricity relation will be set.")
+        return None
+    else:
+        busses = relations.carrier.drop_duplicates().values
+        print(f"Gas electricity relations will be set for busses: {busses}")
+        return relations
+
+
+def get_bpchp_output_parameters(scalars):
+    r"""Gets 'output_parameters' of backpressure CHPs from scalars and
+    returns None if it is missing or None."""
+    bpchp_outs_raw = scalars.loc[
+        (scalars.tech == "bpchp") & (scalars.var_name == "output_parameters")
+    ]
+
+    # drop rows that have empty dict as var_value
+    bpchp_outs = drop_values_by_keyword(bpchp_outs_raw, "{}")
+    if bpchp_outs.empty:
+        return None
+    else:
+        return bpchp_outs
+
+
+def add_output_parameters_to_bpchp(parameters, energysystem):
+    r"""
+    Adds keywords for electricity-gas relation constraint to backpressure CHPs.
+
+    This is necessary as oemof.tabular does not support `output_parameters` of these components,
+    yet. The keywords are set as attributes of the output flow towards `heat_bus`.
+
+    Parameters
+    ----------
+    parameters : pd.DataFrame
+        Contains output_parameters of backpressure CHP scalars.
+    energysystem : oemof.solph.network.EnergySystem
+        The energy system
+    """
+    # rename column 'name' as is collides with iterrows()
+    parameters.rename(columns={"name": "name_"}, inplace=True)
+    for i, element in parameters.iterrows():
+        # get heat bus the component is connected to
+        bus = energysystem.groups[element.name_].heat_bus
+
+        # get keyword and boolean value
+        split_str = element.var_value.split('"')
+        keyword = split_str[1]
+        value = bool(split_str[2].split("}")[0].split()[1])
```

Comment:
@jnnr I actually thought that `load_b3_scalars()` turned the `var_value` of output_parameters and input_parameters into a dictionaries already. But seems like that's not the case. Here it is a string.

Reply:
If this should be done in load_b3_scalars, you can open an issue.